zulipterminal/model.py

```diff
@@ -960,6 +961,43 @@ def get_other_subscribers_in_stream(
                 if sub != self.user_id
             ]
 
+    def _clean_custom_profile_data(
+        self, custom_profile_data: Dict[str, Dict[str, str]]
+    ) -> Dict[str, str]:
+        profile_fields = {}
+        for field in self.initial_data["custom_profile_fields"]:
+            profile_fields[str(field["id"])] = field
+        cleaned_profile_data = {}
+        for field_id in custom_profile_data:
+            value = custom_profile_data[field_id]["value"]
+            field_label = profile_fields[field_id]["name"]
+            field_type = profile_fields[field_id]["type"]
+            if field_type == 3:
+                field_options = json.loads(profile_fields[field_id]["field_data"])
+                field_value = field_options[value]["text"]
+            elif field_type == 6:
+                users = []
+                users_id = list(map(int, value.strip("][").split(",")))
+                for user_id in users_id:
+                    user_data = self._all_users_by_id.get(user_id)
+                    if user_data is not None:
+                        user_name = user_data.get("full_name", "(No name)")
+                        users.append(user_name)
+                field_value = ", ".join(users)
+            elif field_type == 7:
+                field_options = json.loads(profile_fields[field_id]["field_data"])
+                field_subtype = field_options["subtype"]
+                if field_subtype == "github":
+                    field_value = "https://github.com/" + value
+                elif field_subtype == "twitter":
+                    field_value = "https://twitter.com/" + value
+                else:
+                    field_value = value
+            else:
+                field_value = value
```

Comment:
Are 3/6/7 the only ones needing special handling? This will be easier to observe with tests.

Reply:
Yes, from manual testing, it seems so. I'm currently working on adding tests.